it/src/test/java/apoc/it/core/RBACOnLoadTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package apoc.it.core;
+
+import static apoc.util.MapUtil.map;
+import static apoc.util.TestContainerUtil.createEnterpriseDB;
+import static apoc.util.TestContainerUtil.testCall;
+import static apoc.util.TestContainerUtil.testCallEmpty;
+import static java.util.Collections.emptyMap;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import apoc.util.Neo4jContainerExtension;
+import apoc.util.TestContainerUtil;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Session;
+
+public class RBACOnLoadTest {
+
+    private static Neo4jContainerExtension neo4jContainer;
+    private static Driver testUserDriver;
+    private static Session session;
+    private static Session testUserSession;
+    private static final String user = "testUser";
+    private static final String userP = "password1234";
+
+    @BeforeClass
+    public static void beforeClass() {
+        neo4jContainer = createEnterpriseDB(List.of(TestContainerUtil.ApocPackage.CORE), true)
+                .withNeo4jConfig("dbms.memory.heap.max_size", "1GB");
+        neo4jContainer.start();
+
+        assertTrue(neo4jContainer.isRunning());
+        session = neo4jContainer.getSession();
+
+        setupUser();
+        testUserDriver = GraphDatabase.driver(neo4jContainer.getBoltUrl(), AuthTokens.basic(user, userP));
+        testUserDriver.verifyConnectivity();
+        testUserSession = testUserDriver.session();
+    }
+
+    private static void setupUser() {
+        List<String> queries = List.of(
+                "CREATE ROLE test",
+                "CREATE USER " + user + " SET PASSWORD '" + userP + "' SET PASSWORD CHANGE NOT REQUIRED",
+                "GRANT ROLE test, reader, editor TO testUser");
```

Comment:
Reader is a subset of editor, so not sure it provides much value to explicitly grant the reader role

Reply:
Ah yes, I added these at separate parts in my testing and didn't realise! thanks :D